### DISCUSSION TITLE
Help to extract animated replays

### DISCUSSION THREAD

**Discussion by Foreverorneverror** [desuwa/tegaki#11] [General]
First and foremost I should mention that I'm not a programmer and my knowledge of Java is very limited.
I'm trying to get the replays for the images, but without success, always ending up with a blob:null link that has a single png of the last frame.
I assume that my syntax is either entirely incorrect or incomplete:
Tegaki.open({ saveReplay: true,
  onDone: function () { Tegaki.replayRecorder.toBlob()
    var w = window.open(''); 
What's the correct syntax to make a proper replay file (tgkr)? If the syntax is not the issue, then what else could be?
Also as an extension of the previous question, if I'll actually get the proper replay files, should I be able to run them in localhost mode ('https://path/to/replay.tgkr' but as localhost)?

**Comment by desuwa** [comment]
`Tegaki.replayRecorder.toBlob()` gives you a Blob object with the replay data. 

To manually save it as a file you could do something like this inside the onDone callback

```
var replay = Tegaki.replayRecorder.toBlob();
var urlObject = URL.createObjectURL(replay)
var link = document.createElement("a");
link.href = urlObject;
link.download = 'replay.tgkr';
link.click();
```

You should be able to view replays using a local server but it needs to be an actual server and not just a local page you open in a browser.

**Reply by Foreverorneverror** [reply]
Thank you very much! Now it works as intended!
